### PR TITLE
Add tests for MW 1.41, 1.42 and 1.43

### PR DESCRIPTION
**Pull request by paladox**
* Updates PageForm to 5.9 due to issues with 5.3/5.5. 5.9 supports MW 1.39+.
* Fix tests for MW 1.43. Replace PHPUnit_Framework_TestCase with \PHPUnit\Framework\TestCase.

### CODE REVIEW

**Comment by paladox** [comment]
@gesinn-it-gea for some reason it’s stuck on 1.42

stuck on:


--- AfterCreateTablesComplete (Hook) --------------------------------------
Task(s) registered via the hook depend on the functionality implemented and
may take a moment to complete.
-------------------------------------------------------- Import task(s) ---
Importing from smw.search.json …

**Comment by gesinn-it-gea** [comment]
yes, I noticed this as well. Maybe we re-run and see what happens.

**Comment by paladox** [comment]
> yes, I noticed this as well. Maybe we re-run and see what happens.

I did that but seems to have still happened and stuck at the same place.

**Comment by gesinn-it-gea** [comment]
yes got stuck again

**Comment by gesinn-it-gea** [comment]
maybe same as: https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/5675

**Comment by gesinn-it-gea** [comment]
... but the comment in https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/5675 to re-run does not seem a real solution

**Comment by paladox** [comment]
> ... but the comment in [SemanticMediaWiki/SemanticMediaWiki#5675](https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/5675) to re-run does not seem a real solution

Re-trying it for us doesn’t seem to work. I’ve re-tried the test 3 times and it has now gotten stuck 3 times.

**Comment by gesinn-it-gea** [comment]
Maybe PF is causing an issue.

Currently, the Makefile statically installs PageForms v5.5.1 https://github.com/SemanticMediaWiki/SemanticFormsSelect/blob/master/Makefile#L21

I think it would be good to add PF to the CI matrix instead of having a pinned version in the Makefile.

**Comment by paladox** [comment]
@gesinn-it-gea Seems you are correct! https://github.com/SemanticMediaWiki/SemanticFormsSelect/actions/runs/12396929160/job/34606104274?pr=125

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -38,6 +38,27 @@ jobs:
...
+          - mediawiki_version: '1.42'
+            smw_version: dev-master
+            php_version: 8.1
+            database_type: mysql
+            database_image: "mariadb:11.2"
+            coverage: false
+            experimental: false
+          - mediawiki_version: '1.43'
+            smw_version: dev-master
+            php_version: 8.1
```
Comment (gesinn-it-gea) [inline .github/workflows/ci.yml:57]: if we like, we can add "pf_version: xxx" here and to the env belown

Comment (paladox) [reply .github/workflows/ci.yml:57]: 👍

Comment (paladox) [reply .github/workflows/ci.yml:57]: So you want it set for just MW 1.41/42/43? or for all?

Comment (gesinn-it-gea) [reply .github/workflows/ci.yml:57]: If we add it to the matrix I would set it for all matrix cases. If it's the same version for all at the moment, it's okay.

Comment (paladox) [reply .github/workflows/ci.yml:63]: Thanks, done.

Comment (gesinn-it-gea) [reply .github/workflows/ci.yml:63]: awesome!

**Pull request merged**